File: modules/common-net/src/test/java/com/shterneregen/securelan/common/net/transport/SocketCloseTest.java
package com.shterneregen.securelan.common.net.transport;

import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.util.concurrent.atomic.AtomicBoolean;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
import static org.junit.jupiter.api.Assertions.assertTrue;

class SocketCloseTest {
    @Test
    void shouldCloseQuietly() {
        AtomicBoolean closed = new AtomicBoolean(false);

        SocketClose.closeQuietly(() -> closed.set(true));

        assertTrue(closed.get());
    }

    @Test
    void shouldSuppressCloseExceptionsAndAcceptNull() {
        assertDoesNotThrow(() -> SocketClose.closeQuietly(null));
        assertDoesNotThrow(() -> SocketClose.closeQuietly(() -> {
            throw new IOException("boom");
        }));
    }
}

File: modules/common-net/src/test/kotlin/com/shterneregen/securelan/common/net/transport/SocketCloseTest.kt
package com.shterneregen.securelan.common.net.transport

import org.junit.jupiter.api.Assertions.assertDoesNotThrow
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test
import java.io.IOException
import java.util.concurrent.atomic.AtomicBoolean

class SocketCloseTest {
    @Test
    fun shouldCloseQuietly() {
        val closed = AtomicBoolean(false)

        SocketClose.closeQuietly { closed.set(true) }

        assertTrue(closed.get())
    }

    @Test
    fun shouldSuppressCloseExceptionsAndAcceptNull() {
        assertDoesNotThrow { SocketClose.closeQuietly(null) }
        assertDoesNotThrow {
            SocketClose.closeQuietly {
                throw IOException("boom")
            }
        }
    }
}
